src/ds/cli_helper.h

```diff
@@ -19,11 +20,50 @@ namespace cli
     std::string port = {};
   };
 
+  bool parse_address(
+    const std::string_view& addr,
+    ParsedAddress& parsed,
+    const std::string& option_name,
+    const std::string& default_port = "0")
+  {
+    auto found = addr.find_last_of(":");
+    auto hostname = addr.substr(0, found);
+
+    const auto port =
+      found == std::string::npos ? default_port : addr.substr(found + 1);
+
+    // Check if port is in valid range
+    uint16_t port_n;
+    const auto [_, ec] =
+      std::from_chars(port.data(), port.data() + port.size(), port_n);
+    if (ec == std::errc::invalid_argument)
+    {
+      throw CLI::ValidationError(
+        option_name, fmt::format("Port '{}' is not a number", port));
+    }
+    else if (ec == std::errc::result_out_of_range)
+    {
+      throw CLI::ValidationError(
+        option_name,
+        fmt::format("Port '{}'  number is not in range 0-65535", port));
+    }
+    else if (ec != std::errc())
+    {
+      throw CLI::ValidationError(
+        option_name, fmt::format("Error parsing port '{}'", port));
+    }
+
+    parsed.hostname = hostname;
+    parsed.port = port;
+
+    return true;
+  }
+
   CLI::Option* add_address_option(
     CLI::App& app,
     ParsedAddress& parsed,
     const std::string& option_name,
-    const std::string& option_desc,
+    const std::string& option_desc = "",
```

Comment:
Is this right? I would have thought that we always want a non-empty description.

Reply:
Yeah good catch, I think this must have been laziness while adding these options?